app/controllers/playlist_items_controller.rb

```diff
@@ -62,6 +62,41 @@ def update
     render json: { message: "Item was not updated: #{error.message}" }, status: 500 and return
   end
 
+  # GET /playlists/1/items/2/markers
+  def markers
+    @playlist_item = PlaylistItem.find(params['playlist_item_id'])
+    respond_to do |format|
+      format.html do
+        render partial: 'markers', locals: { markers: @playlist_item.marker || [] }
+      end
+    end
+  end
+
+  # GET /playlists/1/items/2/source_details
+  def source_details
+    @playlist_item = PlaylistItem.find(params['playlist_item_id'])
+    respond_to do |format|
+      format.html do
+        render partial: 'current_item'
+      end
+    end
+  end
+
+  # GET /playlists/1/items/2/related_items
+  def related_items
+    @playlist_item = PlaylistItem.find(params['playlist_item_id'])
+    @related_clips = @playlist.related_clips(@playlist_item)
+    respond_to do |format|
+      format.html do
+        unless @related_clips.blank?
```

Comment:
@bkeese Codeclimate is complaining about this duplicate declaration.  Let me know which one you want, or how to adjust and I can add it into my last commit.